backend/src/services/certificate-v3/certificate-v3-service.ts

```diff
@@ -479,9 +637,342 @@ export const certificateV3ServiceFactory = ({
+    }
+
+    const eligibilityCheck = validateRenewalEligibility(originalCert, ca);
+    if (!eligibilityCheck.isEligible) {
+      throw new BadRequestError({
+        message: `Certificate is not eligible for renewal: ${eligibilityCheck.errors.join(", ")}`
+      });
+    }
+
+    if (!internal) {
+      const { permission } = await permissionService.getProjectPermission({
+        actor,
+        actorId,
+        projectId: profile.projectId,
+        actorAuthMethod,
+        actorOrgId,
+        actionProjectType: ActionProjectType.CertificateManager
+      });
+
+      ForbiddenError.from(permission).throwUnlessCan(
+        ProjectPermissionCertificateProfileActions.IssueCert,
+        ProjectPermissionSub.CertificateProfiles
+      );
+    }
+
+    validateCaSupport(ca, "direct certificate issuance");
+
+    const template = await certificateTemplateV2Service.getTemplateV2ById({
+      actor,
+      actorId,
+      actorAuthMethod,
+      actorOrgId,
+      templateId: profile.certificateTemplateId,
+      internal
+    });
+
+    if (!template) {
+      throw new NotFoundError({ message: "Certificate template not found for this profile" });
+    }
+
+    const originalTtlInDays = Math.ceil(
+      (new Date(originalCert.notAfter).getTime() - new Date(originalCert.notBefore).getTime()) / (1000 * 60 * 60 * 24)
+    );
+    const ttl = `${originalTtlInDays}d`;
+
+    const certificateRequest = {
+      commonName: originalCert.commonName || undefined,
+      keyUsages: convertKeyUsageArrayFromLegacy(parseKeyUsages(originalCert.keyUsages)),
+      extendedKeyUsages: convertExtendedKeyUsageArrayFromLegacy(parseExtendedKeyUsages(originalCert.extendedKeyUsages)),
+      subjectAlternativeNames: originalCert.altNames
+        ? originalCert.altNames.split(",").map((san) => {
+            const trimmed = san.trim();
+            const isIp =
+              trimmed.length <= 45 &&
+              (new RE2("^\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}$").test(trimmed) ||
+                new RE2("^([0-9a-fA-F]{1,4}:){7}[0-9a-fA-F]{1,4}$").test(trimmed));
+            return {
+              type: isIp ? CertSubjectAlternativeNameType.IP_ADDRESS : CertSubjectAlternativeNameType.DNS_NAME,
+              value: trimmed
+            };
+          })
+        : [],
+      validity: {
+        ttl
+      }
+    };
+
+    const validationResult = await certificateTemplateV2Service.validateCertificateRequest(
+      profile.certificateTemplateId,
+      certificateRequest
+    );
+
+    if (!validationResult.isValid) {
+      await certificateDAL.updateById(originalCert.id, {
+        renewalError: `Template validation failed: ${validationResult.errors.join(", ")}`
+      });
+
+      throw new BadRequestError({
+        message: `Certificate renewal failed because requested validity period exceeds maximum allowed duration by the profile template: ${validationResult.errors.join(", ")}`
```

Comment:
is this always the case? 

"because requested validity period exceeds maximum allowed duration by the profile template"